### PR TITLE
New Extension: Campeonato Brasileiro

### PR DESCRIPTION
**Pull request by danielsequeira**
## Description

View the latest Campeonato Brasileiro (Brazilian Football League) standings in Raycast.

## Screencast

screenshot-01

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by KoolTheba** [comment]
Hi @danielsequeira 👋 

Thank you very much for the PR for a new Raycast extension about the Campeonato Brasileiro ⚽️ I'm sure people in the community are going to love it!

I'll be reviewing the PR; let me know if you have any questions 🚀

**Review by KoolTheba — CHANGES_REQUESTED** [review]
Hi again @danielsequeira !
I've reviewed the extension and I'd like to add some high level comments for changes that should be done. 
Please let me know anything I could do to help 🙂 or if something is not clear.
I'll leave more specific requests in the code too, thanks your contributions! 🚀
- The app is not following the Raycast extension Command structure. As you are not using Commands, your extension won't work. Here you have some examples on how other extension is built and I'm happy to help anytime: https://developers.raycast.com/examples/todo-list
- You're missing the `.eslintrc.json` configuration file at root level, that extends Raycast eslint rules. Could you please add it to the project? Here you can find an example: https://github.com/raycast/extensions/blob/main/extensions/gitlab/.eslintrc.json
	- After this, you can run the script `npm run lint` to check that your extension is compliance with the Raycast eslint rules, and event `npm run fix-lint` for fixes that the machine is able to do for you.
- Please check out my suggestion about the exposed Token in the code and the documentation for a better token management for Raycast extensions. I highly recommend you to rotate that token and keep it private.
- On regards to the screenshot in the metadata folder, the required size is 2000 x 1250 pixels, could you please upload one screenshot with those dimensions? It's important as is how your cool extension will be seen by the community in the Store. More info: https://developers.raycast.com/information/manifest#command-properties
- README file:
	- Please, could add a bit more information about the extension features, how can be used, and at least a couple of screenshots of the comands extension, which I assume are Top Scorers and Standings 
	- Also, if the user needs to add the token for using the API of Campeonato Brasileiro you mentioned in the README, it's important to add a bit more of information about how to use it while using the extension. You can get the info from the documentation too

File: extensions/campeonato-brasileiro/src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+
+const footballDataEndpoint = "https://api.football-data.org/v4";
+
+export const get_standings = async (): Promise<Table[]> => {
+  const config: AxiosRequestConfig = {
+    method: "get",
+    url: `${footballDataEndpoint}/competitions/BSA/standings`,
+  };
+  config.headers = {
+    "X-Auth-Token": "ce068fce0e994e3293417577749fac15",
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/src/api/index.ts:14]: This is the token I mentioned in my previous comments. Please use the [Raycast Preferences API](https://developers.raycast.com/api-reference/preferences) for managing tokens like this one. Please let me know if I can help 🚀

File: extensions/campeonato-brasileiro/src/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  }
+};
+
+export const getTopScorers = async (): Promise<TopScorer[]> => {
+  const config: AxiosRequestConfig = {
+    method: "get",
+    url: `${footballDataEndpoint}/competitions/BSA/scorers`,
+  };
+  config.headers = {
+    "X-Auth-Token": "ce068fce0e994e3293417577749fac15",
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/src/api/index.ts:36]: This is the Token also should be refactored following the guidelines of my previous comment.

**Comment by danielsequeira** [comment]
Thanks for the review @KoolTheba. On your comments:

> The app is not following the Raycast extension Command structure. As you are not using Commands, your extension won't work. Here you have some examples on how other extension is built and I'm happy to help anytime: https://developers.raycast.com/examples/todo-list

I'm sorry but I don't really understand what I need to change here or what is not working. Can you add some comments to my code regarding this and with suggestions and how it should be changed?

> Please check out my suggestion about the exposed Token in the code and the documentation for a better token management for Raycast extensions. I highly recommend you to rotate that token and keep it private.

That makes sense, however I didn't find your suggestion in the code? Where is this? And how can I use a token in a Raycast extension without exposing it?


Regarding the other changes I will go ahead and do them.

File: extensions/campeonato-brasileiro/src/api/index.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    method: "get",
+    url: `https://example.org/analytics?text=${text}`,
+  };
+
+  try {
+    await axios(config);
+  } catch (e) {
+    return;
+  }
+};
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/src/api/index.ts:64]: [Raycast policies](https://developers.raycast.com/basics/prepare-an-extension-for-store#analytics) does not allowed third party apps for analytics. Soon, we will add support to give developers more insights into how their extension is being used.

File: extensions/campeonato-brasileiro/package.json [hunk 4]
```diff
@@ -0,0 +1,53 @@
+{
+  "$schema": "https://www.raycast.com/schemas/extension.json",
+  "name": "campeonato-brasileiro",
+  "title": "Campeonato brasileiro",
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/package.json]:
```suggestion
  "title": "Campeonato Brasileiro",
```
I'm adding a code suggestion to the extension title for using Title Cased. [Here](https://developers.raycast.com/basics/prepare-an-extension-for-store#extensions-and-commands-naming) you can find more info.
After the eslint config file installation, you'd have found this by running `npm run lint`

File: extensions/campeonato-brasileiro/package.json [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+  "commands": [
+    {
+      "name": "standings",
+      "title": "Standings",
+      "description": "View the latest Campeonato Brasileiro standings.",
+      "mode": "view"
+    },
+    {
+      "name": "topScorers",
+      "title": "View Top scorers",
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/package.json]:
```suggestion
      "title": "View Top Scorers",
```
Same here with the Title convention 🙂

**Review by KoolTheba — CHANGES_REQUESTED** [review]
Hi @danielsequeira !
Thanks a lot for reviewing this comments and make changes, good job! 🚀

**Comment by KoolTheba** [comment]
Thank you for the changes @danielsequeira 🚀
Let me reply specifically to your questions:
> I'm sorry but I don't really understand what I need to change here or what is not working. Can you add some comments to my code regarding this and with suggestions and how it should be changed?
> 
I'll add more comments in the code, for sure!

> That makes sense, however I didn't find your suggestion in the code? Where is this? And how can I use a token in a Raycast extension without exposing it?
> 
Apologies for that! Here you can find the comment about the token: https://github.com/danielsequeira/extensions/pull/1#discussion_r1192366278
You can find also the same targeted comment about the Tokens in the same file.
Is there anything else I could do to clarify, please let me know.

**Comment by danielsequeira** [comment]
All clear and all the changes should be made.

File: extensions/campeonato-brasileiro/.eslintrc.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
...
+    "node": true
+  },
+  "parser": "@typescript-eslint/parser",
+  "plugins": ["@typescript-eslint"],
+  "extends": [
+    "eslint:recommended",
+    "plugin:@typescript-eslint/recommended",
+    "prettier"
+  ]
+}
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/.eslintrc.json:14]:
```suggestion
{
  "root": true,
  "extends": ["@raycast"]
}
```
In order to make your extension the most compatible as possible with Raycast, please use this rules extension that the Raycast team recommended.

Comment (danielsequeira) [reply extensions/campeonato-brasileiro/.eslintrc.json:14]: I did this, but when I ran `npm run lint`, I got this error:

> error -  run ESLint
Oops! Something went wrong! :(

ESLint: 7.32.0

ESLint couldn't find the config "@raycast" to extend from. Please check that the name of the config is correct.

The config "@raycast" was referenced from the config file in "/Users/danielsequeira/workspace/danielsequeira-extensions/extensions/campeonato-brasileiro/.eslintrc.json".

If you still have problems, please stop by https://eslint.org/chat/help to chat with the team.


That is why I just reverted to those other plugins.

File: extensions/campeonato-brasileiro/src/standings.tsx [hunk 7]
```diff
@@ -0,0 +1,81 @@
+import { Action, ActionPanel, Icon, Image, List } from "@raycast/api";
+import { useState } from "react";
+import useStandings from "./hooks/useStandings";
+
+export default function GetStandings() {
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/src/standings.tsx:5]:
```suggestion
export default function Command() {
```
Here is what I meant about the Commands. I'm adding some suggestion here; the name of the file should match the command name in the manifest, which you did perfect.

File: extensions/campeonato-brasileiro/src/topScorers.tsx [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+
+function getCountryDescription(countryName: string | undefined) {
+  if (countryName == undefined) {
+    return "?";
+  }
+
+  return countryName + " " + flag(countryName);
+}
+
+export default function GetTopScorers() {
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/src/topScorers.tsx:27]:
```suggestion
export default function Command() {
```
This is also about what I mentioned about Commands.

**Review by KoolTheba — CHANGES_REQUESTED** [review]
I hope this help to clarify what I meant about the Commands structure for Raycast extensions. Please let me know if you need additional help.

File: extensions/campeonato-brasileiro/.eslintrc.json (hunk 6 above)
Comment (KoolTheba) [reply extensions/campeonato-brasileiro/.eslintrc.json:14]: Yes, great catch @danielsequeira !
You need to install this dependency: `@raycast/eslint-config` by `npm i -D @raycast/eslint-config`
Apologies for that! 🙏

**Comment by danielsequeira** [comment]
@KoolTheba last requested changes pushed

File: extensions/campeonato-brasileiro/README.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# Campeonato Brasileiro
+
+View Campeonato Brasileiro standings and top scorers.
+
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/README.md:4]: Please @danielsequeira include the screenshots so the community can see how beautiful is your extension in the Store and also a bit more of the cool features that you provided for this extension 🚀

File: extensions/campeonato-brasileiro/README.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
...
+## How to get Football Data API Key
+
+1. Open https://www.football-data.org/client/register
+2. Create an account by providing a name and email address
+3. Copy the API Key you received by email
```
Comment (KoolTheba) [inline extensions/campeonato-brasileiro/README.md:9]: Could you also add an screenshot, when the Preferences are in place in the code, about how the user must include the token to make the extension work?
Thank you!

**Comment by KoolTheba** [comment]
Great job @danielsequeira 🚀
We're getting super close to ship it! 
I've spotted other few little things (typos, etc) in the code that I'll provide asap.
Thanks for your contributions and for being part of the Raycast community!

**Comment by danielsequeira** [comment]
@KoolTheba images added to readme

**Pull request closed without merging**